microceph/ceph/nfs.go

```diff
@@ -0,0 +1,230 @@
+package ceph
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/canonical/microceph/microceph/constants"
+	"github.com/canonical/microceph/microceph/database"
+	"github.com/canonical/microceph/microceph/interfaces"
+)
+
+// EnableNFS enables the NFS Ganesha service on the cluster and adds initial configuration.
+func EnableNFS(s interfaces.StateInterface, nfs *NFSServicePlacement, monitorAddresses []string) error {
```

Comment:
It would be great if we could make use of the lxd revert module here. This would allow us to rollback changes to the system and cluster in case of faults.

E.g. [similar to the AddOSD() func](https://github.com/canonical/microceph/blob/4910cfbd937860690fba764bd9c85ace5707f969/microceph/ceph/osd.go#L531)